File: src/main/java/com/omkar/smartnotes/service/NoteService.java
package com.omkar.smartnotes.service;

import com.omkar.smartnotes.entity.Note;

import java.util.List;

import com.omkar.smartnotes.dto.NoteRequest;

public interface NoteService {

    Note createNote(NoteRequest request);

    List<Note> getAllNotes();

    void deleteNote(Long id);

    Note getNoteById(Long id);

    Note updateNote(Long id, NoteRequest request);

}

File: frontend/src/services/noteService.js
import api from "./api";

export const getAllNotes = async () => {
    const response = await api.get("/api/notes");
    return response.data;
};

export const getNoteById = async (id) => {
    const response = await api.get(`/api/notes/${id}`);
    return response.data;
};

export const createNote = async (note) => {
    const response = await api.post("/api/notes", note);
    return response.data;
};

export const updateNote = async (id, note) => {
    const response = await api.put(
        `/api/notes/${id}`,
        note
    );

    return response.data;
};

export const deleteNote = async (id) => {
    await api.delete(`/api/notes/${id}`);
};